src/cpp/include/openvino/genai/whisper_pipeline.hpp

```diff
@@ -17,6 +17,22 @@ using OptionalWhisperGenerationConfig = std::optional<WhisperGenerationConfig>;
 
 using RawSpeechInput = std::vector<float>;
 
+struct OPENVINO_GENAI_EXPORTS WhisperDecodedResultChunk {
+    // start of chunk in seconds
+    float start_ts;
+
+    // end of chunk in seconds
+    // -1.0f if chunk started but model did not predict an ending timestamp
+    // can happen if audio is cut off in the middle of a word
+    float end_ts = -1.0f;
+    std::string text;
+};
+
+class OPENVINO_GENAI_EXPORTS WhisperDecodedResults : public DecodedResults {
+public:
+    std::optional<std::vector<WhisperDecodedResultChunk>> chunks = std::nullopt;
```

Comment:
why results can be optional?

Reply:
`return_timestamps` can be `false`. In that case we could return no chunks (`DecodedResults`),empty chunks, null chunks. I thought null chunks are more explicit than empty vector, ie no intent to return timestamps at all. There is a case when `return_timestamps = true` but chunks intended to be empty.
We could also return something like `std::variant<DecodedResults, WithTimestampsDecodedResults>` from `generate` (no chunks option). But I'm not sure if it's a good practice in c++. HF doesn't return 'chunks' key in case of `return_timestamps = false`.